### PR TITLE
demonstrate that R.isEmpty({}) evaluates false

### LINKED ISSUE #1194: R.isEmpty({}) returns false
```
R.isEmpty('') returns true,
R.isEmpty([]) returns true
```

but

`R.isEmpty({}) returns false`

=> If an object is observed as a collection of (enumerable) properties, this can be considered inconsistent. If `Object.keys({})` returns `[]`, it may make sense to consider the object as empty

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #1194

### CODE REVIEW

File: src/isEmpty.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ var _curry1 = require('./internal/_curry1');
  *      R.isEmpty([]); //=> true
  *      R.isEmpty(''); //=> true
  *      R.isEmpty(null); //=> false
+ *      R.isEmpty({}); //=> false
```
Comment (buzzdecafe) [inline src/isEmpty.js]: should we annotate this possibly surprising example to point out, e.g.:

`R.isEmpty(null); //=> false; {} is not a list.`

Comment (davidchambers) [inline src/isEmpty.js]: For the `R.isEmpty({})` example, do you mean?

Comment (buzzdecafe) [inline src/isEmpty.js]: yes, for the commented line

Comment (davidchambers) [inline src/isEmpty.js]: You wrote `R.isEmpty(null)` which confused me. :stuck_out_tongue_winking_eye:

Comment (buzzdecafe) [inline src/isEmpty.js]: oops! fixed

Comment (davidchambers) [inline src/isEmpty.js]: I've updated the pull request. It now includes `R.isEmpty({length: 0})` as an example.

File: src/isEmpty.js [hunk 2]
```diff
@@ -12,10 +12,13 @@ var _curry1 = require('./internal/_curry1');
...
- *      R.isEmpty([1, 2, 3]); //=> false
- *      R.isEmpty([]); //=> true
- *      R.isEmpty(''); //=> true
- *      R.isEmpty(null); //=> false
+ *      R.isEmpty([1, 2, 3]);   //=> false
+ *      R.isEmpty([]);          //=> true
+ *      R.isEmpty('');          //=> true
+ *      R.isEmpty(null);        //=> false
+ *      R.isEmpty(R.keys({}));  //=> true
+ *      R.isEmpty({});          //=> false ({} does not have a length property)
```
Comment (buzzdecafe) [inline src/isEmpty.js]: "`{}` is not list, i.e. not array-like" might be better here. i think specifying that it needs a `length` property is an implementation detail we should keep hidden.

File: src/isEmpty.js [hunk 3]
```diff
@@ -12,10 +12,13 @@ var _curry1 = require('./internal/_curry1');
...
+ *      R.isEmpty({length: 0}); //=> true
```
Comment (buzzdecafe) [inline src/isEmpty.js]: and then here, maybe: "('{length: 0}` is array-like)"

Comment (davidchambers) [inline src/isEmpty.js]: Would we then state the following?

```javascript
R.isEmpty('');          //=> true ('' is array-like)
```

I don't think of strings as array-like.

For the record I remain in favour of removing this function. It's just sugar for `R.propEq('length', 0)`, but this fact is not obvious.

Comment (buzzdecafe) [inline src/isEmpty.js]: >  It's just sugar for `R.propEq('length', 0)`

put that way, i think i agree

Comment (buzzdecafe) [inline src/isEmpty.js]: > I don't think of strings as array-like.

now that more & more list functions are being used for strings, we may have to revisit that question. at present, String is excluded from `isArrayLike`

**Comment by davidchambers** [comment]
I'll merge this as it's an improvement on what we have now and I'd like to include this change in v0.15.1 (which I'm about to publish).

**Pull request merged**